planner/core/src/main/java/eu/seaclouds/platform/planner/core/Planner.java

```diff
@@ -167,17 +174,19 @@ public String rePlan() throws ParsingException, IOException {
         List<NameValuePair> optParams = getReqParams(
                 new BasicNameValuePair("aam", aam),
                 new BasicNameValuePair("dam", dam),
-                new BasicNameValuePair("mmResults", mmOutput)
+                new BasicNameValuePair("offers", mmOutput)
         );
-        String optResult = optimizerClient.postRequest("/optimize", optParams);
-        return optResult; //TODO: check the output for the optimizer
+
+        Pair<String, String> optResult = optimizerClient.postRequest("/optimize", optParams);
+        if(optResult.first.equals("200")){
+            return optResult.second; //TODO: check the output for the optimizer
+        }
+        return null; //TODO: return failure
     }
 
     public String generateDam() {
```

Comment:
If the optimizer returns a String (why a json? I think planner wants to use the optimizer as a lib, not as a service?) I think Optimizer can be improved as well.
Anyway, `Planner.generateDam()` output doesn't need to be a String because Optimizer.optimize output is a String, don't you think?